README.md

```diff
@@ -46,6 +49,97 @@ In your template you can do
 {{user.twitter}}
 ```
 and it will populate once the full user is loaded
+
+## How to use
+If you haven't customized any of your adapters, serializers or store, the addon will work out of the box. By default is uses JSONAPISerializer and JSONAPIAdapter.
+
+If you made any customization, you will need to include proper mixins. For adapter:
+
+```js
+// app/adapters/application.js
+
+import DS from 'ember-data';
+import PartialModelAdapter from 'ember-data-partial-model/mixins/adapter';
+const { RESTAdapter } = DS;
+
+export default RESTAdapter.extend(PartialModelAdapter);
+
+```
+
+For serializer (both RESTSerializer and JSONAPISerializer are available):
+
+```js
+// app/adapters/serializer.js
+
+import DS from 'ember-data';
+import PartialModelRESTSerializer from 'ember-data-partial-model/mixins/rest-serializer';
+const { RESTSerializer } = DS;
+
+export default RESTSerializer.extend(PartialModelRESTSerializer);
+```
+
+or
+
+```js
+// app/adapters/serializer.js
+
+import DS from 'ember-data';
+import PartialModelJSONAPISerializer from 'ember-data-partial-model/mixins/jsonapi-serializer';
+const { JSONAPISerializer } = DS;
+
+export default JSONAPISerializer.extend(PartialModelJSONAPISerializer);
+```
+
+
+For store:
+
+```js
+// app/services/store.js
+
+import DS from 'ember-data';
+import PartialModelStore from 'ember-data-partial-model/mixins/store';
+const { Store } = DS;
+
+export default Store.extend(PartialModelStore, {});
+```
+
+## Supporting custom serializers
+If you are using different serializer than REST or JSONAPI, you can create custom mixin to support it, you will just need to provide some functions for handling logic specific to your serializer. The basic layout is the following:
+
+```js
+import Ember from 'ember';
+import parentSerializerMixin from 'ember-data-partial-model/mixins/serializer';
+const { Mixin } = Ember;
+
+export default Mixin.create(parentSerializerMixin, {
+  serializeBelongsTo: function(snapshot, json, relationship) {
+     // overwrite this function to handle serialization of
+     // the relationship for extension, most likely you want to
+     // return null for partial extension. Required.
+     //  if (relationship.options.isPartialExtension) {
+     //   return null;
+     //  } else {
+     //    return this._super(...arguments);
+     //  }
+     //
+  },
+
+  _normalizePartialRelationship: function(modelClass, hash, descriptor) {
+    // add this function to handle normalization of partial extension as a relationship. Required.
+  },
+
+  _copyAttributesFromPartialToParent: function(ourHash, partialData) {
+    // add this function to handle copying attributes from partial extension to top-level hash. Required.
+  },
+
+  _modifyHashBasedOnDescriptorKey: function(ourHash, descriptorKey) {
```

Comment:
The base serializer is already defining this function so their's no need to define it unless you need to do something with it.